src/renderer/coremods/contextMenu/index.tsx

```diff
@@ -88,68 +95,134 @@ type ContextMenuData = ContextMenuProps["ContextMenu"] & {
  * @internal
  * @hidden
  */
-export function _insertMenuItems(menu: ContextMenuData): void {
+export function _buildPatchedMenu(menu: ContextMenuData): React.ReactElement | null {
   const { navId } = menu;
+  const {
+    MenuGroup,
+    ContextMenu,
+  }: {
+    MenuGroup: React.FC<ContextMenuProps["MenuGroup"]>;
+    ContextMenu: React.FC<ContextMenuProps["ContextMenu"] & { plugged?: boolean }>;
+  } = ContextComponents;
+
+  //return nothing as we weren't able to get ContextMenu component, gets handled in plain text patch
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (!ContextMenu || menu.plugged) return null;
 
   // No items to insert
-  if (!menuItems[navId]) return;
-
-  // Already inserted items
-  // If this isn't here, another group of items is added every update
-  if (menu.plugged) return;
-
-  // We delay getting the items until now, as importing at the start of the file causes Discord to hang
-  // Using `await import(...)` is undesirable because the new items will only appear once the menu is interacted with
-  const { MenuGroup } = getByProps<Record<string, React.ComponentType>>([
-    "Menu",
-    "MenuItem",
-    "MenuGroup",
-  ]) || { MenuGroup: undefined };
-  if (!MenuGroup) return;
+  // Or MenuGroup Component is not available
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (!menuItems[navId] || !MenuGroup) return <ContextMenu {...menu} plugged={true} />;
 
   // The data as passed as Arguments from the calling function, so we just grab what we want from it
   const data = menu.data[0];
 
-  const repluggedGroup = <MenuGroup />;
-  repluggedGroup.props.id = "replugged";
-  repluggedGroup.props.children = [];
-
-  // Add in the new menu items right above the DevMode Copy ID
-  // If the user doesn't have DevMode enabled, the new items will be at the bottom
-  if (!Array.isArray(menu.children)) menu.children = [menu.children];
-  const hasCopyId = menu.children
-    .at(-1)
-    ?.props?.children?.props?.id?.startsWith("devmode-copy-id-");
-  if (hasCopyId) {
-    menu.children.splice(-1, 0, repluggedGroup);
+  // make children array if it's not already
+  if (!Array.isArray(menu.children)) {
+    menu.children = [menu.children];
+    if (!Array.isArray(menu.children)) {
+      return <ContextMenu {...menu} plugged={true} />;
+    }
+  }
```

Comment:
Why are you checking if it's an array after making it an array?

Reply:
i dont remember. maybe i will remove it then